File: kumoh-code-2017/Java/#1 기본/#2 수업/소스코드/11. 은행관리 프로그램 (완성본)/은행관리/FreeAccount.java
package 은행관리;

public class FreeAccount extends Account
{
	private long money;
	
	public FreeAccount(int account_number, int pin, long amount)
	{
		super(account_number, pin, amount);
	}
    
	public void credit(long amount)
	{
		money = (long)(this.getBalance() + (amount * 1.02));
		
		plusBalance(money);
	}
	
    public void debit(long amount) 
    {
		money = (long)(this.getBalance() - (amount * 1.02));
		
		minusBalance(money);
    }
}

File: kumoh-code-2017/C++/#2 설계/과제15 은행관리 프로그램/FreeAccount.cpp
#include"FreeAccount.h"

FreeAccount::FreeAccount(int accountNumber, int password, int balance) :Account(accountNumber, password, balance){
	
}

 //====메소드 재정의 (다형성) =====

void FreeAccount::credit(int money){
	balance = balance+ money;
}

void FreeAccount::debit(int money){ 
	Screen screen;
	if ((balance-money) < 0){
		screen.displayMsg("잔고가 0원 미만입니다.(출금을 취소합니다.)");
	}
	else{
		balance = balance - money;
	}
}